components/prize_pool/commons/prize_pool_import.lua

```diff
@@ -0,0 +1,502 @@
+			end
+
+			-- Opponents knocked out from sole section (se) or lower bracket (de)
+			if coordinates.sectionIndex == #bracket.sections
+
+				-- Include opponents directly knocked out from the upper bracket
+				or firstDeRoundIndex and coordinates.roundIndex < firstDeRoundIndex then
+
+				table.insert(groupKeys, {2, coordinates.depth, 2})
+			end
+
+			return groupKeys
+		end
+	end
+
+	local placementEntries = {}
+	for matchId in MatchGroupCoordinates.dfs(bracket) do
+		for _, groupKey in ipairs(getGroupKeys(matchId)) do
+			table.insert(placementEntries, {
+				groupKey = groupKey,
+				matchId = matchId,
+				matchPlacement = groupKey[3],
+			})
+		end
+	end
+
+	Array.sortInPlaceBy(placementEntries, function(entry)
+		return Array.extend(
+			entry.groupKey,
+			bracket.coordinatesByMatchId[entry.matchId].matchIndexInRound
+		)
+	end)
+
+	return Array.groupBy(placementEntries, function(entry)
+		return table.concat(entry.groupKey, '.')
+	end)
+end
+
+-- Returns the semifinals match IDs of a bracket if the losers also play in a
+-- third place match to determine placement.
+function Import._getPreTiebreakMatchIds(bracket)
+	local firstBracketData = bracket.bracketDatasById[bracket.rootMatchIds[1]]
+	local thirdPlaceMatchId = firstBracketData.thirdPlaceMatchId
+
+	local sfMatchIds = {}
+	if thirdPlaceMatchId and bracket.matchesById[thirdPlaceMatchId] then
+		for _, lowerMatchId in ipairs(firstBracketData.lowerMatchIds) do
+			sfMatchIds[lowerMatchId] = true
+		end
+	end
+	return sfMatchIds
+end
+
+-- Finds the first round in where upper bracket opponents drop to the lower
+-- bracket. Returns nil if it cannot be determined unambiguously, or if the
+-- bracket is not double elimination.
+function Import._findDeRoundIndex(bracket)
+	if #bracket.sections ~= 2 then
+		return nil
+	end
+	local countsByRound = MatchGroupCoordinates.computeRawCounts(bracket)
+
+	for roundIndex = 1, #bracket.rounds do
+		local lbCount = countsByRound[roundIndex][2]
+		if lbCount == 0 then
+			return roundIndex
+		elseif lbCount > 0 then
+			return nil
+		end
+	end
+end
+
+function Import._mergePlacements(lpdbEntries, placements)
+	for placementIndex, lpdbPlacement in ipairs(lpdbEntries) do
+		placements[placementIndex] = Import._mergePlacement(
+			lpdbPlacement,
+			placements[placementIndex] or Import._emptyPlacement(placements[placementIndex - 1], #lpdbPlacement)
+		)
+	end
+
+	return placements
+end
+
+function Import._emptyPlacement(priorPlacement, placementSize)
+	priorPlacement = priorPlacement or {}
+	local placement = Table.deepCopy(priorPlacement, {copyMetatable = true})
+
+	local placeStart = (priorPlacement.placeEnd or 0) + 1
+	local placeEnd = (priorPlacement.placeEnd or 0) + placementSize
+
+	return Table.mergeInto(placement, {
+			args = {},
+			hasUSD = false,
+			opponents = {},
+			prizeRewards = {},
+			placeStart = placeStart,
+			placeEnd = placeEnd,
+			placeDisplay = Import._getPlaceDisplay(placeStart, placeEnd),
```

Comment:
this is needed, because this is only set for the case that we import into an empty placement, so all the stuff that would get set by the prizepool slot reading isn't set
this includes the place display
if we remove this it will just error